### ISSUE TITLE
`find_package(range-v3)` with CMake 3.10 does not work (range-v3 0.10.0)

### ISSUE THREAD

**Issue by Marc--Olivier** [ericniebler/range-v3#1411]
## Description

When I run CMake 3.10 on a project that includes range-v3 0.10.0,
I get the following error message:

```
Cannot specify link libraries for target "range-v3::meta" which is not built by this project.
```

It took me quite some time to realize that CMake 3.10 does not seem able to
process [cmake/range-v3-config.cmake](https://github.com/ericniebler/range-v3/blob/0.10.0/cmake/range-v3-config.cmake). At some point, I tried out with CMake 3.16.2 and it worked. It would be good to update range-v3-config.cmake to fail if the CMake version is not greater than, for example, 3.16 (I did not take the time to figure out the smallest CMake version that works).

Note that we could use a check similar to the check contained in the generated file `lib/cmake/range-v3/range-v3-targets.cmake` when installing the library:

```
if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 2.5)
   message(FATAL_ERROR "CMake >= 2.6.0 required")
endif()
```

## Steps to reproduce:

1. Compile range-v3 and install it:
```sh
git clone https://github.com/ericniebler/range-v3 --branch=0.10.0 --depth=1
mkdir install
mkdir build && pushd build
cmake ../range-v3 -G Ninja -DCMAKE_BUILD_TYPE=Release -DRANGE_V3_TESTS=OFF -DCMAKE_INSTALL_PREFIX="../install"
ninja
ninja install
popd
```

2. Generate a CMake project:
```sh
mkdir test_project && pushd test_project
cat >CMakeLists.txt <<EOL
cmake_minimum_required(VERSION 3.10)
find_package(range-v3 HINTS "../install")
EOL
cmake .
```

=> I get the following error:

```sh
CMake Error at /tmp/tmp.V7kykEykmj/install/lib/cmake/range-v3/range-v3-config.cmake:5 (target_link_libraries):
  Cannot specify link libraries for target "range-v3::meta" which is not
  built by this project.
Call Stack (most recent call first):
  CMakeLists.txt:2 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/tmp.V7kykEykmj/test_project/CMakeFiles/CMakeOutput.log".
```

**Comment by marehr** [comment]
Seems to be this change f75dac8a0e23aea257b5a26a1afa5e23c94b841c that was introduced by the PR #1385.

To quote eric:

> Thanks. I hope this really (really) fixes it this time.

Seems (really) not :laughing: 

BTW cmake 3.11.4 and upwards is working fine and I could reproduce your error with cmake 3.10.3

I personally think that you should always use the most recent version of cmake, because it is extremely easy to install.

**Comment by marehr** [comment]
This patch should work on cmake 3.10 and below:

```patch
diff --git a/cmake/range-v3-config.cmake b/cmake/range-v3-config.cmake
index 8a43da6..775c147 100644
--- a/cmake/range-v3-config.cmake
+++ b/cmake/range-v3-config.cmake
@@ -2,6 +2,12 @@ include("${CMAKE_CURRENT_LIST_DIR}/range-v3-targets.cmake")
 add_library(range-v3::meta INTERFACE IMPORTED)
 add_library(range-v3::concepts INTERFACE IMPORTED)
 add_library(range-v3::range-v3 INTERFACE IMPORTED)
+if(CMAKE_VERSION VERSION_LESS 3.11) # workaround for target_link_libraries on lower cmake versions
+set_target_properties(range-v3::meta PROPERTIES INTERFACE_LINK_LIBRARIES "range-v3-meta")
+set_target_properties(range-v3::concepts PROPERTIES INTERFACE_LINK_LIBRARIES "range-v3-concepts")
+set_target_properties(range-v3::range-v3 PROPERTIES INTERFACE_LINK_LIBRARIES "range-v3")
+else()
 target_link_libraries(range-v3::meta INTERFACE range-v3-meta)
 target_link_libraries(range-v3::concepts INTERFACE range-v3-concepts)
 target_link_libraries(range-v3::range-v3 INTERFACE range-v3)
+endif()
```

**Comment by johelegp** [comment]
I didn't consider the minimum required version when making the change. The [patch notes for 3.11](https://cmake.org/cmake/help/latest/release/3.11.html#commands) says that `target_link_libraries()` was added then. @marehr, could you open a PR, please?

**Comment by marehr** [comment]
@johelegp done, see #1412

**Comment by johelegp** [comment]
Thank you.

**Comment by johelegp** [comment]
This can be closed.

**Comment by marehr** [comment]
@Marc--Olivier Can you verify that this fixes it for you and if so close this?

**Comment by Marc--Olivier** [comment]
I tested 0b2aef9bbeb0a25710ddbacb2568eb7ea0415404 with CMake 3.10 , and did not get the error message anymore in my project, thanks!

BTW, when I created the ticket, I had already updated cmake to 3.16, so I was not impacted by this issue anymore. I just thought that other people may spend time figuring out that they are not using the correct version of CMake, that's why I created the issue.

In the ticket description, I suggested to get a better error message (related to CMake), but making CMake 3.10 work is even better! 👍 😄
